File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/trending/model/TrendingPerson.java
package dev.reuss.tmdb.domain.trending.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;
import dev.reuss.tmdb.value.media.MediaType;

import java.util.List;

/**
 * Person returned by TMDB people trending.
 *
 * @param adult              whether the person is marked as adult
 * @param id                 TMDB person id
 * @param name               person name
 * @param originalName       original person name
 * @param mediaType          media type, usually {@code person}
 * @param popularity         person popularity
 * @param gender             gender value returned by TMDB
 * @param knownForDepartment known department
 * @param profilePath        profile image path
 * @param knownFor           known movie or TV results
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TrendingPerson(
        boolean adult,

        int id,

        String name,

        @JsonProperty("original_name")
        String originalName,

        @JsonProperty("media_type")
        MediaType mediaType,

        double popularity,

        int gender,

        @JsonProperty("known_for_department")
        String knownForDepartment,

        @JsonProperty("profile_path")
        String profilePath,

        @JsonProperty("known_for")
        List<TrendingPersonKnownFor> knownFor
) implements TmdbModel {
    public TrendingPerson {
        knownFor = TmdbCollections.list(knownFor);
    }

}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/trending/model/TrendingPerson.kt
package dev.reuss.tmdb.domain.trending.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.value.media.MediaType

/**
 * Person returned by TMDB people trending.
 *
 * @property adult whether the person is marked as adult
 * @property id TMDB person id
 * @property name person name
 * @property originalName original person name
 * @property mediaType media type, usually `person`
 * @property popularity person popularity
 * @property gender gender value returned by TMDB
 * @property knownForDepartment known department
 * @property profilePath profile image path
 * @property knownFor known movie or TV results
 */
@JvmRecord
data class TrendingPerson(
    @all:JsonProperty("adult")
    val adult: Boolean,

    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("name")
    val name: String?,

    @all:JsonProperty("original_name")
    val originalName: String?,

    @all:JsonProperty("media_type")
    val mediaType: MediaType?,

    @all:JsonProperty("popularity")
    val popularity: Double,

    @all:JsonProperty("gender")
    val gender: Int,

    @all:JsonProperty("known_for_department")
    val knownForDepartment: String?,

    @all:JsonProperty("profile_path")
    val profilePath: String?,

    @all:JsonProperty("known_for")
    val knownFor: List<TrendingPersonKnownFor> = emptyList()
) : TmdbModel